### PR TITLE
docs: how redpanda works

### LINKED ISSUE #970: How Redpanda works
This covers the architecture content in https://github.com/redpanda-data/documentation/issues/865. (The sizing content remains in https://github.com/redpanda-data/documentation/pull/890.)

This adds a new intro page about how Redpanda works based on https://go.redpanda.com/redpanda-sizing-guidelines. In the next sprint, the beginning part of this page (Redpanda features) will move into the new introduction to Redpanda page.

### PR DESCRIPTION
**Pull request by micheleRP**
Resolves #970 

This PR covers the architecture content in #865. (The sizing content remains in https://github.com/redpanda-data/documentation/pull/890.) 

This adds a new intro page about how Redpanda works based on https://go.redpanda.com/redpanda-sizing-guidelines. In the next sprint, the beginning part of this page (Redpanda features) will move into the new introduction to Redpanda page. 

Review deadline: Thu Dec 22

preview:
- https://deploy-preview-964--redpanda-documentation.netlify.app/docs/platform/introduction/architecture/
- https://deploy-preview-964--redpanda-documentation.netlify.app/docs/platform/introduction/

### CODE REVIEW

**Comment by micheleRP** [comment]
@jrkinley, @Feediver1, @dcoded: Please review/approve this PR. I'd like to merge it this week, before our winter break! This PR covers the architecture content in https://github.com/redpanda-data/documentation/issues/865. (The sizing content remains in https://github.com/redpanda-data/documentation/pull/890.)

This adds a new intro page about how Redpanda works based on https://go.redpanda.com/redpanda-sizing-guidelines. In the next sprint, the beginning part of this page (Redpanda features) will move into the new Introduction to Redpanda page.

File: docs/platform/introduction/architecture.mdx [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+title: How Redpanda Works
+---
+
+<head>
+    <meta name="title" content="Redpanda's Architecture | Redpanda Docs"/>
+    <meta name="description" content="Redpanda is a distributed system that is composed of multiple nodes (servers) and data producers and consumers (clients)."/>
+</head>
+
+Redpanda provides mission-critical data infrastructure for streaming real-time data. It's fast, it's easy, and it keeps your data safe.
```
Comment (Feediver1) [inline docs/platform/introduction/architecture.mdx:10]:
```suggestion
Redpanda provides mission-critical data infrastructure for streaming real-time data. It's fast, easy, and it keeps your data safe. 
```

File: docs/platform/introduction/architecture.mdx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+## Redpanda features
+
+### Fast
+  
+Redpanda is designed from the ground up for maximum performance on any data streaming workload. It can scale up to use all available resources on a single machine and scale out to distribute performance across multiple machines. This enables new use cases that require high throughput and low latency, with a minimal hardware footprint.
+
+### Simple
+  
+Redpanda is packaged as a single binary. It does not rely on any external systems. It’s easy to install, configure, and operate in any production environment. Redpanda is fully compatible with the Apache Kafka© API, so it works with the ecosystem of tools and integrations that have been built on the Kafka protocol. It can be deployed in your data center or in the cloud on bare metal, containers, and virtual machines.
```
Comment (Feediver1) [inline docs/platform/introduction/architecture.mdx:20]: It’s easy to install, configure, and operate in any production environment.

^^ This is from the blog? Reads like marketing speak, and in my experience, product docs typically refrain from saying something is "easy"...I'd remove this sentence.

File: docs/platform/introduction/architecture.mdx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+### Reliable
+  
+At its core, Redpanda is a durable and fault-tolerant transaction log for storing data streams. It uses the Raft consensus algorithm to write data to log files and replicate that data across multiple servers. Raft facilitates communication between the nodes in a Redpanda cluster to make sure that they agree on changes and remain in sync, even if a minority of them are in a failure state. This enables Redpanda to ride out partial environmental failures to deliver a predictable performance, even at high loads.
```
Comment (Feediver1) [inline docs/platform/introduction/architecture.mdx:24]: Maybe link "Raft consensus algorithm" to: https://raft.github.io/

File: docs/platform/introduction/architecture.mdx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+## Redpanda architecture
+
+### Producers, consumers, and topics
+
+Redpanda is a distributed system composed of multiple nodes (servers) and data producers and consumers (clients). Producers are client applications that write data to Redpanda in the form of events. An event is simply a record of something changing state, like a credit card transaction, new coordinates of a taxi, or someone pressing play on a TV series. Consumers read these events from Redpanda for processing and forwarding on to downstream systems.
```
Comment (Feediver1) [inline docs/platform/introduction/architecture.mdx:30]:
```suggestion
Redpanda is a distributed system composed of multiple nodes (servers), and data producers and consumers (clients). Producers are client applications that write data to Redpanda in the form of events. An event is simply a record of something changing state, like a credit card transaction, new coordinates of a taxi, or someone pressing play on a TV series. Consumers read these events from Redpanda for processing and forwarding on to downstream systems.
```

File: docs/platform/introduction/architecture.mdx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+Redpanda implements a thread-per-core programming model through its use of the [Seastar](https://seastar.io/) library. This allows Redpanda to pin each of its application threads to a CPU core to avoid context switching and blocking. It combines this with structured message passing (SMP) to asynchronously communicate between the pinned threads. This allows Redpanda to avoid the overhead of context switching and expensive locking operations to improve processing performance and efficiency.
+
+From a sizing perspective, Redpanda’s ability to efficiently use all available hardware lets it scale up to get the most out of your infrastructure, before you’re forced to scale out to meet the demands of your workload. Redpanda delivers better performance with a smaller footprint, resulting in reduced operational costs and complexity.
+
```
Comment (Feediver1) [inline docs/platform/introduction/architecture.mdx:81]: How about a **What's Next** heading with links to Redpanda Licensing and Install rpk?

**Review by Feediver1 — APPROVED** [review]
See comments/suggestions.

File: docs/platform/introduction/architecture.mdx (hunk 5 above)
Comment (micheleRP) [reply docs/platform/introduction/architecture.mdx:81]: Thanks! I added links to Redpanda Licensing and Install Redpanda.

File: docs/platform/introduction/architecture.mdx (hunk 1 above)
Comment (micheleRP) [reply docs/platform/introduction/architecture.mdx:10]: @Feediver1 ah, let me know if this suggestion is important to you. I like the original better. It reminds me of Alex drilling this in every all-hands.

**Comment by micheleRP** [comment]
@jrkinley: Could you please review & approve this PR? I'd like to merge it Friday, Dec 23. This PR covers the architecture content in https://github.com/redpanda-data/documentation/issues/865. (The sizing content remains in https://github.com/redpanda-data/documentation/pull/890.)
 In the next sprint, the beginning part of this page (Redpanda features) will move into the new Introduction to Redpanda page.

**Comment by JakeSCahill** [comment]
When I wanted to understand how Redpanda works, I watched some great YouTube videos where Alex explains the details. Is it worth linking to those too?

File: docs/platform/introduction/architecture.mdx [hunk 6]
```diff
@@ -0,0 +1,85 @@
+---
+title: How Redpanda Works
+---
+
+<head>
+    <meta name="title" content="Redpanda's Architecture | Redpanda Docs"/>
+    <meta name="description" content="Redpanda is a distributed system that is composed of multiple nodes (servers) and data producers and consumers (clients)."/>
```
Comment (daisukebe) [inline docs/platform/introduction/architecture.mdx:7]: > Redpanda is a distributed system that is composed of multiple nodes (servers) and data producers and consumers (clients)

Is this statement good? @Feediver1 
Redpanda itself doesn't contain clients, either producers or consumers.

File: docs/platform/introduction/architecture.mdx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+</head>
+
+Redpanda provides mission-critical data infrastructure for streaming real-time data. It's fast, it's easy, and it keeps your data safe. 
+
+## Redpanda features
```
Comment (Feediver1) [inline docs/platform/introduction/architecture.mdx:12]:
```suggestion
## Redpanda differentiators
```

**Comment by micheleRP** [comment]
Closing this PR, because it was easier for me to move this content into #866, since some of the content overlapped. I also think it'll be easier for reviewers with those pages together.

**Pull request closed without merging**